src/test/scala/kamon/graphite/GraphiteSenderSpec.scala

```diff
@@ -0,0 +1,164 @@
+        rangeSamplers = List.empty
+      )
+      val testee = new TestGraphiteSender(senderConfig)
+
+      //act
+      testee.reportPeriodSnapshot(periodSnapshot)
+
+      //assert
+      testee.messages shouldBe List("kamon-graphiteprefix.custom_user_counter.count;tag1=111 42 1517000993\n")
+    }
+
+    "send gauge metrics" in {
+      //arrange
+      val periodSnapshot = PeriodSnapshot(
+        from, to,
+        counters = List.empty,
+        gauges = List(MetricSnapshotBuilder.gauge("jvm.heap-size", TagSet.Empty, 150000000)),
+        histograms = List.empty,
+        timers = List.empty,
+        rangeSamplers = List.empty
+      )
+      val testee = new TestGraphiteSender(senderConfig)
+
+      //act
+      testee.reportPeriodSnapshot(periodSnapshot)
+
+      //assert
+      testee.messages shouldBe List("kamon-graphiteprefix.jvm_heap-size.value;tag1=111 1.5E8 1517000993\n")
+    }
+
+    "send histograms" in {
+      //arrange
+      val periodSnapshot = PeriodSnapshot(
+        from, to,
+        counters = List.empty,
+        gauges = List.empty,
+        histograms = List(MetricSnapshotBuilder.histogram("my.histogram", TagSet.Empty)(1, 2, 4, 6)),
+        timers = List.empty,
+        rangeSamplers = List.empty
+      )
+      val testee = new TestGraphiteSender(senderConfig)
+
+      //act
+      testee.reportPeriodSnapshot(periodSnapshot)
+
+      //assert
+      testee.messages shouldBe List(
+        "kamon-graphiteprefix.my_histogram.count;tag1=111 4 1517000993\n",
+        "kamon-graphiteprefix.my_histogram.min;tag1=111 1 1517000993\n",
+        "kamon-graphiteprefix.my_histogram.max;tag1=111 6 1517000993\n",
+        "kamon-graphiteprefix.my_histogram.p0.5;tag1=111 1 1517000993\n",
+        "kamon-graphiteprefix.my_histogram.p0.9;tag1=111 1 1517000993\n",
+        "kamon-graphiteprefix.my_histogram.p0.999;tag1=111 1 1517000993\n",
+        "kamon-graphiteprefix.my_histogram.average;tag1=111 3 1517000993\n",
+        "kamon-graphiteprefix.my_histogram.sum;tag1=111 13 1517000993\n")
+    }
+
+    "send timers" in {
+      //arrange
+      val periodSnapshot = PeriodSnapshot(
+        from, to,
+        counters = List.empty,
+        gauges = List.empty,
+        histograms = List.empty,
+        timers = List(MetricSnapshotBuilder.histogram("my.timer", TagSet.Empty)(1, 2, 4, 6)),
+        rangeSamplers = List.empty
+      )
+      val testee = new TestGraphiteSender(senderConfig)
+
+      //act
+      testee.reportPeriodSnapshot(periodSnapshot)
+
+      //assert
+      testee.messages shouldBe List(
+        "kamon-graphiteprefix.my_timer.count;tag1=111 4 1517000993\n",
+        "kamon-graphiteprefix.my_timer.min;tag1=111 1 1517000993\n",
+        "kamon-graphiteprefix.my_timer.max;tag1=111 6 1517000993\n",
+        "kamon-graphiteprefix.my_timer.p0.5;tag1=111 1 1517000993\n",
+        "kamon-graphiteprefix.my_timer.p0.9;tag1=111 1 1517000993\n",
+        "kamon-graphiteprefix.my_timer.p0.999;tag1=111 1 1517000993\n",
```

Comment:
Just a small comment on the test details: in the reference configuration, the percentiles are `[50.0, 90.0, 99.0]`, but in this test we have `[.5, .9, .999]` which are not really the same, would be nice to have the tests run with the default configuration as well!

Reply:
good find!
adapted test to use the same values as the default config - makes far more sense.